lib/ecto/changeset.ex

```diff
@@ -174,20 +174,54 @@ defmodule Ecto.Changeset do
   ## Schemaless changesets
 
   In the changeset examples so far, we have always used changesets to
-  validate and cast data contained in a struct, such as the `%User{}`
-  struct defined by the `User` module.
-
-  However, changesets can also be used with data in a plain map, by
-  passing a tuple containing both the data and the supported types:
-
-      data  = %{}
+  validate and cast data contained in a struct defined by an Ecto Schema, such as the `%User{}`
+  struct defined by the `User` module. 
+  
+  However, changesets can also be used without defining our structs using an Ecto schema:
```

Comment:
What do you think about?

```
However, changesets can also be used with data in a plain map, by
passing a tuple containing both the data and the supported types:

    data  = %{}
    types = %{name: :string}
    params = %{name: "Callum"}

    changeset =
      {data, types}
      |> Ecto.Changeset.cast(params, Map.keys(types))
      |> Ecto.Changeset.validate_required(:name)

Elixir structs are just bare maps underneath. For this reason, changesets
can also be used with any struct:

    defmodule User do
      defstruct [:name, :email, :age]
    end 

    user = %User{}
    types = %{name: :string, email: :string, age: :integer}
    params = %{name: "Callum", email: "callum@example.com"}

    changeset =
      {user, types}
      |> Ecto.Changeset.cast(params, Map.keys(types))
      |> Ecto.Changeset.validate_required(:email)

Such functionality makes Ecto extremely useful to cast, validate and prune
data even if it is not meant to be persisted to the database.
```

Note also that I change the `types` variable to match the ones in the `User` struct. I've also indented the examples with 4 spaces, so they are rendered as code :)

Reply:
The change in structure offers a much more readable improvement. I would further add doctest `iex` statements to the examples, just as a written proof to readers "This code is guaranteed to work" but I'm wasn't too sure how well that would play with your repository.

Reply:
I think you would have to give the structs literal definitions in order for that to work, because I don't think doctest allows you to define Elixir structs directly in your code.